### PR TITLE
record pid of exec'd process

### PR DESCRIPTION
**Pull request by runcom**
Resurrecting docker/docker#19095 as per @LK4D4 comment docker/docker#18635 (comment). Would be nice for docker to expose exec's pids in order to give an admin full control over execs (e.g. someone spawns run-away processes like tail -f /dev/null which never ends and he wants to kill it directly).

ping @thaJeztah @LK4D4 @mlaventure @crosbymichael

### CODE REVIEW

File: integration-cli/docker_api_exec_test.go [hunk 1]
```diff
@@ -69,6 +69,10 @@ func (s *DockerSuite) TestExecAPIStart(c *check.C) {
 	id := createExec(c, "test")
 	startExec(c, id, http.StatusOK)
 
+	var execJSON struct{ PID int }
+	inspectExec(c, id, &execJSON)
+	c.Assert(execJSON.PID, checker.GreaterOrEqualThan, 0)
```
Comment (mlaventure) [inline integration-cli/docker_api_exec_test.go]: `> 1` ?

**Review by mlaventure — CHANGES_REQUESTED** [review]
One nit, but LGTM.

File: integration-cli/docker_api_exec_test.go (hunk 1 above)
Comment (runcom) [inline integration-cli/docker_api_exec_test.go]: right

Comment (runcom) [inline integration-cli/docker_api_exec_test.go]: Fixed

**Review by mlaventure — APPROVED** [review]

**Comment by mlaventure** [comment]
LGTM.

File: docs/reference/api/docker_remote_api_v1.25.md [hunk 2]
```diff
@@ -3286,7 +3286,8 @@ Return low-level information about the `exec` command `id`.
         "tty": true,
         "user": "1000"
       },
-      "Running": false
+      "Running": false,
+      "Pid": "42000"
```
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.25.md]: before I forget, or someone is "trigger happy" and merges; can you add this to the API changes for 1.25; https://github.com/docker/docker/blob/master/docs/reference/api/docker_remote_api.md#v125-api-changes

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: docs/reference/api/docker_remote_api_v1.25.md (hunk 2 above)
Comment (runcom) [inline docs/reference/api/docker_remote_api_v1.25.md]: @thaJeztah thx!

**Comment by runcom** [comment]
@thaJeztah fixed your comments, any other doc changes needed?

File: libcontainerd/client_windows.go [hunk 3]
```diff
@@ -264,12 +264,12 @@ func (clnt *client) Create(containerID string, checkpoint string, checkpointDir
 
 // AddProcess is the handler for adding a process to an already running
 // container. It's called through docker exec.
```
Comment (thaJeztah) [inline libcontainerd/client_windows.go]: Should we update the description of this function to mention it returns the PID?

File: libcontainerd/client_linux.go [hunk 4]
```diff
@@ -30,17 +30,17 @@ type client struct {
 	liveRestore   bool
 }
 
-func (clnt *client) AddProcess(ctx context.Context, containerID, processFriendlyName string, specp Process) error {
+func (clnt *client) AddProcess(ctx context.Context, containerID, processFriendlyName string, specp Process) (int, error) {
```
Comment (thaJeztah) [inline libcontainerd/client_linux.go]: Can you copy the GoDoc from the windows file?

**Review by thaJeztah — COMMENTED** [review]
just a few nits, but looks good otherwise

**Comment by runcom** [comment]
@thaJeztah fixed

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐸

**Pull request merged**